bootstrap/src/hash.c

```diff
@@ -0,0 +1,49 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <unistd.h>
+#include <caml/mlvalues.h>
+#include <caml/alloc.h>
+
+// Return number of entropy bits written to buf.
+static size_t
+hemlock_getentropy(uint8_t *buf, size_t nbits_in) {
+  size_t nbytes_out = (nbits_in + 7) / 8;
+  size_t nbits_out = nbytes_out * 8;
+
+  return (getentropy((void *)buf, nbytes_out) == 0) ? nbits_out : 0;
+}
+
+static value
+value_of_result(const char *arg) {
+  uint64_t entropy = *((uint64_t *)arg);
+
+  caml_copy_int64(entropy);
+}
+
+// The OCaml FFI is quite low-level, such that returning (Int64.t option) is a
+// lot of trouble relative to returning an array.  As a consequence this
+// function returns an empty array to indicate error.
+//
+//                                       nbits -> [|entropy...|]
+// val hemlock_hash_state_entropy_nbits: usize -> Int64.t array
```

Comment:
This may look over-engineered, but it will soon support 128-bit seeding for hashing based on [MurmurHash3](https://github.com/aappleby/smhasher) without modification.